### ISSUE TITLE
Generate spans for remote calls invoked via System.Net.WebRequest (or sub-classes)

### ISSUE THREAD

**Issue by chetanmangsuli84** [DataDog/dd-trace-dotnet#701]
Currently we are unable to instrument any http or https requests in asp.net. For eg: when we do `Response.Redirect` or 

```csharp
HttpWebRequest request = (HttpWebRequest)WebRequest.Create(txtinput.Text);
HttpWebResponse response = (HttpWebResponse)request.GetResponse();
```

We are not able to instrument this `HttpRequest` containing `txtinput.Text`.

**Comment by macrogreg** [comment]
Internal tracker:
https://datadoghq.atlassian.net/browse/DOTNET-209

**Comment by macrogreg** [comment]
Internal tracker:
https://datadoghq.atlassian.net/browse/DOTNET-209

**Comment by zacharycmontoya** [comment]
I wanted to post an update that in .NET Tracer v1.21.1, we've ensured that calls to `WebRequest.GetResponse()` made to `System 2.0.0.0` are properly instrumented. If this does not solve the original issue, please let me know. I expect the compiler to generate virtcall's to `WebRequest.GetResponse` whenever the C# code is making a `GetResponse()` call on an `HttpWebRequest`, so we should be covering the above scenario.

**Comment by feinoujc** [comment]
@zacharycmontoya we've upgrade to 1.21.1 but we still have some missing spans as described in #869

**Comment by sowee15** [comment]
We are also using RestSharp's ExecuteAsync in some cases (which uses Begin/End methods) and we also don't see those http calls in our datadog traces (but we see traces for non async calls). Can we expect instrumentation of those async calls to work in the future? Thanks!

**Comment by ssanyaltb** [comment]
Per @sowee15 's comment, we'd like to know when logging for asynchronous calls will be supported for the RestSharp ExecuteAsync scenarios, Thanks

**Comment by stevenaw** [comment]
This would be helpful for us as well, specifically the callback-based async methods such as `BeginGetResponse` / `EndGetResponse` and associated stream closing.

**Comment by sowee15** [comment]
Any idea if/when this will be implemented? Thank you!

**Comment by lucaspimentel** [comment]
Thank you for reporting this issue. We added automatic instrumentation for methods `HttpWebRequest.BeginGetResponse` and `HttpWebRequest.EndGetResponse` in #1658 which was included in [release 1.28.6](https://github.com/DataDog/dd-trace-dotnet/releases/tag/v1.28.6).

If this does not resolve your issue, please contact support@datadoghq.com and reference this GitHub issue so we can further assist you. Thanks!

**Comment by stevenaw** [comment]
Thanks!
